### PR TITLE
moving AssetProcessorServerUnitTest out of legacy framework

### PR DESCRIPTION
**Pull request by AMZN-stankowi**
moving AssetProcessorServerUnitTest out of legacy framework

Signed-off-by: AMZN-stankowi <4838196+AMZN-stankowi@users.noreply.github.com>

### CODE REVIEW

File: Code/Tools/AssetProcessor/native/unittests/AssetProcessorServerUnitTests.cpp [hunk 1]
```diff
@@ -34,172 +39,243 @@
...
+
+            // We need to process all events, since AssetProcessorServer is also on the same thread
+            while (numberOfConnection.load() && !failureOccurred)
+            {
+                QCoreApplication::sendPostedEvents(0, QEvent::DeferredDelete);
+                QCoreApplication::processEvents();
+            }
 
-        UNIT_TEST_CHECK(failureOccurred == false);
+            EXPECT_TRUE(failureOccurred == false);
```
Comment (amzn-mike) [inline Code/Tools/AssetProcessor/native/unittests/AssetProcessorServerUnitTests.cpp:195]: Shouldn't this be expect_false?

File: Code/Tools/AssetProcessor/native/unittests/AssetProcessorServerUnitTests.cpp [hunk 2]
```diff
@@ -34,172 +39,243 @@
...
+
+                AZStd::this_thread::sleep_for(AZStd::chrono::milliseconds(numTimeWait + idx));
+                numberOfConnection.fetch_sub(1);
+
+                if (connection.NegotiationFailed() != failNegotiation)
+                {
+                    failureOccurred = true;
+                }
+
+                EXPECT_TRUE(connection.NegotiationFailed() == failNegotiation);
```
Comment (amzn-mike) [inline Code/Tools/AssetProcessor/native/unittests/AssetProcessorServerUnitTests.cpp:168]: Should probably make this EXPECT_EQ

**Review by jackalbe — APPROVED** [review]

File: Code/Tools/AssetProcessor/native/unittests/AssetProcessorServerUnitTests.cpp (hunk 1 above)
Comment (AMZN-stankowi) [reply Code/Tools/AssetProcessor/native/unittests/AssetProcessorServerUnitTests.cpp]: fixed

File: Code/Tools/AssetProcessor/native/unittests/AssetProcessorServerUnitTests.cpp (hunk 2 above)
Comment (AMZN-stankowi) [reply Code/Tools/AssetProcessor/native/unittests/AssetProcessorServerUnitTests.cpp]: fixed

**Review by amzn-mike — APPROVED** [review]

**Pull request merged**